src/commands/dev/index.ts

```diff
@@ -98,6 +98,19 @@ class UpProcessManager {
       });
     });
 
+    let recreated_socket = false;
+    this.server.on('error', (e: any) => {
+      if (e.code === 'EADDRINUSE' && this.server && !recreated_socket) {
+        // Socket already exists (likely from a previous run that didnt get cleaned up properly)
+        // Remove it and listen again creating a new socket.
+        fs.rmSync(this.socket);
+        this.server.listen(this.socket);
+        recreated_socket = true;
```

Comment:
This should be the first thing in the function. No idea if `listen` calls an sync function or checks for the pipe beforehand. Also not sure if it changes per version of node which knowing Node would not be surprising.

Reply:
I did now quickly check back to node 12.x and it seems to be a very similar description to the latest version docs https://nodejs.org/docs/latest-v12.x/api/net.html#net_server_listen so I do think it this particular bit of functionality is consistent across node versions 😬 (at least, Listen was still async in previous versions)

I think I understand what you're saying about first thing in the function but just to clarify, what we probably should do is:
```typescript
this.server = net.createSever();
this.server.on('connection', ...);
this.server.on('error', ...);
this.server.listen();
```

so that we don't hit this error before we actually start listening, right? (Listen is async which is why this worked on my machine, but I will definitely change this - let me know if you meant something else 😛 )

Reply:
What I more mean is that `listen` could use `fs.readFileSync` to check that the file exists. If for some reason the cleanup code does not work due to some change then the `recreated_socket` will never hit true so your loop protection would not work. Instead, we should mark we have just been in the function once so that no matter what happens we just exit if we ever hit this function again.

Reply:
Oooooh, I understand what you're saying now, brain is working at half speed today but that makes complete sense lol, I'll move this to the top of the block